test/test_visualization.jl

```diff
@@ -60,6 +60,8 @@ isdir(outdir) && rm(outdir, recursive=true)
   @testset "plot recipes" begin
     pd = PlotData2D(sol)
 
+    # Add dummy method implementation to avoid errors when Plots.jl is not loaded
+    RecipesBase.is_key_supported(k::Symbol) = true
```

Comment:
That's some serious type piracy and changes the behavior of RecipesBase.jl permanently within the session running this. Additionally, this part doesn't really test the interface we would like to test (see my other comment here).

Reply:
Arrrrrrrr! :pirate_flag: :skull: Yes, yes, I knew this wouldn't sit well. Originally I didn't want to include Plots in the test dependencies for performance reasons, but I'll write down the timings and see what happens.